zrml/prediction-markets/src/lib.rs

```diff
@@ -1012,35 +1048,87 @@ mod pallet {
             }
         }
 
-        fn is_market_active(end: MarketEnd<T::BlockNumber>) -> bool {
+        fn ensure_create_market_end(end: MarketEnd<T::BlockNumber>) -> DispatchResult {
             match end {
                 MarketEnd::Block(block) => {
                     let current_block = <frame_system::Pallet<T>>::block_number();
-                    current_block < block
+                    ensure!(current_block < block, Error::<T>::EndBlockTooSoon);
                 }
                 MarketEnd::Timestamp(timestamp) => {
                     let now = T::Timestamp::now();
-                    now < timestamp.saturated_into()
+                    ensure!(now < timestamp.saturated_into(), Error::<T>::EndTimestampTooSoon);
+                }
+            };
+            Ok(())
+        }
+
+        fn ensure_disputes_does_not_exceed_max_disputes(num_disputes: u32) -> DispatchResult {
```

Comment:
#don't trust the compiler :smile: 
```suggestion
        #[inline]
        fn ensure_disputes_does_not_exceed_max_disputes(num_disputes: u32) -> DispatchResult {
```